swiper.el

```diff
@@ -693,39 +693,43 @@ WND, when specified is the window."
           ;; RE can become an invalid regexp
           (while (and (ignore-errors (re-search-forward re end t))
                       (> (- (match-end 0) (match-beginning 0)) 0))
-            (unless (and (consp ivy--old-re)
-                         (null
-                          (save-match-data
-                            (ivy--re-filter ivy--old-re
-                                            (list
-                                             (buffer-substring-no-properties
-                                              (line-beginning-position)
-                                              (line-end-position)))))))
-              (let ((mb (match-beginning 0))
-                    (me (match-end 0)))
-                (unless (> (- me mb) 2017)
-                  (swiper--add-overlay mb me
-                                       (if (zerop ivy--subexps)
-                                           (cadr swiper-faces)
-                                         (car swiper-faces))
-                                       wnd 0))))
-            (let ((i 1)
-                  (j 0))
-              (while (<= (cl-incf j) ivy--subexps)
-                (let ((bm (match-beginning j))
-                      (em (match-end j)))
-                  (when (and (integerp em)
-                             (integerp bm))
-                    (while (and (< j ivy--subexps)
-                                (integerp (match-beginning (+ j 1)))
-                                (= em (match-beginning (+ j 1))))
-                      (setq em (match-end (cl-incf j))))
-                    (swiper--add-overlay
-                     bm em
-                     (nth (1+ (mod (+ i 2) (1- (length swiper-faces))))
-                          swiper-faces)
-                     wnd i)
-                    (cl-incf i)))))))))))
+            ;; Don't highlight a match if it goes across
+            ;; multiple lines.
+            (unless (> (line-number-at-pos (match-end 0))
+                       (line-number-at-pos (match-beginning 0)))
```

Comment:
Why not
```el
(when (> 2 (count-lines (match-beginning 0) (match-end 0))) ...)
```
or similar?

Reply:
I didn't know about `count-lines`. But that actually seems less clear to me. One has to figure out why `2` instead of `1`, and the docstring does not seem all too obvious to me:

```
Return number of lines between START and END.
This is usually the number of newlines between them,
but can be one more if START is not equal to END
and the greater of them is not at the start of a line.
```

I think using `line-number-at-pos` makes it obvious that what we are checking is if the beginning and ending are on different lines.

Reply:
Actually, `line-number-at-pos` is super slow, since it recomputes the line number from the beginning of the buffer. Checking if the match has a newline should be faster.

Reply:
> But that actually seems less clear to me. One has to figure out why `2` instead of `1`, and the docstring does not seem all too obvious to me

In these cases it is customary to
* add a comment explaining the logic and/or write a helper function with a clearer docstring
* look up `count-lines` in the Elisp manual, e.g. via <kbd>C-h</kbd><kbd>S</kbd>`count-lines`<kbd>RET</kbd>, where the return value is explained in greater detail
* `report-emacs-bug` suggesting to clarify the docstring of `count-lines`

> Actually, `line-number-at-pos` is super slow, since it recomputes the line number from the beginning of the buffer.

Yes, which is why I suggested calling `count-lines` directly.

> Checking if the match has a newline should be faster.

This is also a possibility, but are you sure that:
1. constructing a `match-string` is faster than `count-lines` (the latter is backed by a fast newline cache)
2. checking for carriage returns, as `count-lines` does when `selective-display` is enabled, is never required

?